hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUpgradeActionsModule.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.hadoop.ozone.recon;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Singleton;
+import org.apache.hadoop.ozone.recon.upgrade.InitialConstraintUpgradeAction;
+
+/**
+ * Guice module for binding upgrade actions as singletons in Recon.
+ */
+public class ReconUpgradeActionsModule extends AbstractModule {
+
+  @Override
+  protected void configure() {
+    // Bind UpdateUnhealthyContainersConstraintAction as a singleton
+    bind(InitialConstraintUpgradeAction.class).in(Singleton.class);
```

Comment:
This need not be registed / singleton, as used once only, can argument be passed ?